docs/proposals/kubelet-systemd.md

```diff
@@ -0,0 +1,181 @@
+
+* the `kubelet` detects the `.service` unit from which it was launched and uses that for monitoring its own resource usage
+
+If the `kubelet` is using the `docker` container runtime, the following behavior is proposed:
+
+* the `kubelet` detects the `.service` unit from which the `docker` daemon was launched (i.e. `docker.service`)
+
+If the `kubelet` wants to set memory limits on the `docker` daemon, it should only do this via a flag:
+
+* `--container-runtime-memory-threshold-limit="70": if specified, the kubelet will enforce memory limits on the container runtime agent`
+
+If specified, the `kubelet` should interface with the `systemd` API to set runtime scoped limits on the `docker.service` directly.  If not
+specified, the `kubelet` will delegate to whatever value was set directly via the administrator that configured the `docker.service`.
+
+Finally, to monitor system resource usage, the `kubelet` should accept the following flag:
+
+* `--system-slices="system.slice": Comma-delimited set of slices that the kubelet will monitor for system usage when using systemd cgroup driver`
+
+By default, it is assumed that the `.service` files that define host-processes `kubelet`, `kube-proxy`, and `docker` are in the `system.slice`.
+
+This means that by default, the `kubelet` can monitor host daemon usage via monitoring of the enumerated slice easily.
+
+The `kubelet` will verify that the proper `cpu` and `memory` controllers are enabled on the enumerated slice.
+
+## kubelet cgroup-root under systemd
+
+The `kubelet` supports a `cgroup-root` flag which is the optional root cgroup to use for pods.  It is handled on a best-effort basis by the
+container runtime.
+
+For this proposal, if the `kubelet` is launched with `--cgroup-driver=systemd`, the parameter is expected to take the form of a `.slice` name.
+
+For example, if the following is specified `--cgroup-driver=systemd --cgroup-root=kubernetes.slice`, the `kubelet` will make a best-effort attempt
+to parent all pod containers under the `kubernetes.slice`.
+
+This feature is important to facilitate local node accounting that distinguishes between system processes and pod processes.
+
+For example, if you use `docker` in your OS distro, you may choose to run your own host containers managed separately from the `kubelet`, and those
+should be treated like system processes.  By default, they will go in the `system.slice`.  For this reason, `docker` features that depend on a daemon-wide default setting (https://github.com/docker/docker/pull/18379) are insufficient for Kubernetes needs.  Instead, its required that the `kubelet` is always able to specify a `cgroup-parent` on `systemd` on a per container basis (https://github.com/opencontainers/runc/pull/336)
+
+## kubelet cgroups per quality of service (qos) tier
+
+It is desired to parent containers in a cgroup based on their qos tier to facilitate local node accounting scenarios.
```

Comment:
That's fine , hope idea was clear for future discussion topics.  I just wanted to relate with cgroup-root that we have today.